presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -257,7 +260,19 @@ private Date getDate(int columnIndex, DateTimeZone localTimeZone)
         }
 
         try {
-            return new Date(DATE_FORMATTER.withZone(localTimeZone).parseMillis(String.valueOf(value)));
+            LocalDate localDate = DATE_FORMATTER.parseLocalDate(String.valueOf(value));
+            if (modernDate(localDate.getYear())) {
+                long millis = localDate.toDateTimeAtStartOfDay(localTimeZone).getMillis();
```

Comment:
Let's keep the optimistic path as simple as possible.
Previously it was
```
return new Date(DATE_FORMATTER.withZone(localTimeZone).parseMillis(String.valueOf(value)));
```

now it can be
```
long millis = DATE_FORMATTER.withZone(localTimeZone).parseMillis(String.valueOf(value));
if (millis >= MODERN_YEAR_MILLIS) {
    return new Date(millis);
}
```